File: services/ctl-api/internal/pkg/interests/defaults.go
package interests

// AllEvents returns the new-subscription default: a sentinel Interests config
// that matches every supported lifecycle and approval event. This is what
// the "Send all events" toggle in the picker writes when it's checked, and
// what new webhooks/slack subs default to.
func AllEvents() Interests {
	return Interests{AllEvents: true}
}

// Default returns the materialised "power user opted out of AllEvents"
// baseline. Five resources (installs, stacks, components, sandboxes,
// install_configurations) are present with empty ops (= all sub-ops),
// Outcome=Completion, and approval flags both true (where supported).
// Runners + actions are absent (= off).
//
// Toggling "Send all events" off in the picker writes this exact shape so
// users land on a sensible per-resource starting point instead of an empty
// config that silently drops every event.
func Default() Interests {
	return Interests{
		Resources: map[ResourceKind]ResourceCfg{
			ResourceInstalls: {
				Outcome:           OutcomeCompletion,
				ApprovalRequests:  true,
				ApprovalResponses: true,
			},
			ResourceStacks: {
				Outcome: OutcomeCompletion,
			},
			ResourceComponents: {
				Outcome:           OutcomeCompletion,
				ApprovalRequests:  true,
				ApprovalResponses: true,
				DriftDetected:     true,
			},
			ResourceSandboxes: {
				Outcome:           OutcomeCompletion,
				ApprovalRequests:  true,
				ApprovalResponses: true,
				DriftDetected:     true,
			},
			ResourceInstallConfigurations: {
				Outcome:           OutcomeCompletion,
				ApprovalRequests:  true,
				ApprovalResponses: true,
			},
		},
	}
}

File: services/dashboard-ui/client/components/interests/defaults.ts
// Mirrors AllEvents() and Default() in
// services/ctl-api/internal/pkg/interests/defaults.go.

import type { Interests } from './types'

// New-subscription default: matches every supported lifecycle + approval event.
export const allEvents = (): Interests => ({ all_events: true })

// Power-user opted out of AllEvents baseline. Four resources present, runners +
// actions absent. Empty ops = all sub-ops; OutcomeCompletion + both approval
// flags true. drift_detected is on for the two resources that can produce a
// drift_detected event (components, sandboxes — see RESOURCES_WITH_DRIFT_DETECTED).
export const defaultInterests = (): Interests => ({
  resources: {
    installs: { outcome: 'completion', approval_requests: true, approval_responses: true },
    stacks: { outcome: 'completion' },
    components: { outcome: 'completion', approval_requests: true, approval_responses: true, drift_detected: true },
    sandboxes: { outcome: 'completion', approval_requests: true, approval_responses: true, drift_detected: true },
    install_configurations: { outcome: 'completion', approval_requests: true, approval_responses: true },
    app_branches: { outcome: 'completion', config_synced: true },
  },
})

export const isZero = (i: Interests): boolean =>
  !i.all_events && (!i.resources || Object.keys(i.resources).length === 0)
